files/build_templates/sonic_debian_extension.j2

```diff
@@ -246,6 +246,16 @@ sudo dpkg --root=$FILESYSTEM_ROOT -i $debs_path/sonic-utilities-data_*.deb || \
 # in bash.bashrc, so we copy a version of the file with it enabled here.
 sudo cp -f $IMAGE_CONFIGS/bash/bash.bashrc $FILESYSTEM_ROOT/etc/
 
+# Remove 'PyGObject' package installed via apt
+sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y remove python3-gi
```

Comment:
>python3-gi [](start = 85, length = 10)

Do you know which step install `python3-gi` ?

Reply:
`python3-gi` is installed as part of `software-properties-common` installation in build_debian.sh
https://github.com/Azure/sonic-buildimage/blob/eeb955489f7818ea2c21b86cbece3db2c76f5df5/build_debian.sh#L196-L200

Reply:
Then I am confused, if installed in sonic-buildimage/build_debian.sh, we need to clean up in that file.
You mentioned "Remove 'PyGObject' package installed via apt", but I did not see the relationship.

---
In reply to: [568892398](https://github.com/Azure/sonic-buildimage/pull/6273#discussion_r568892398) [](ancestors = 568892398)

Reply:
`PyGObject`  python package is part of `python3-gi` and its installation via apt leads to ' PyGObject' not being installed via pip, while installing ‘sonic-host-services’. But ‘python3-gi’ is removed while ‘apt-get autoremove’ is executed during the final stages of the build and thus apt version of ‘PyGObject’ also gets uninstalled.

To avoid the above behavior, ‘python3-gi’ is removed before installing ‘sonic-host-services’ to install ‘PyGObject’ via pip itself.
It was placed here, since its removal was only required for installing 'sonic-host-services'. 
Let me know if 'python3-gi' removal has to be placed in build_debian.sh, before invoking sonic_debian_extension.sh.

Reply:
You reply shows great insight, really appreciate!
My understanding of build_debian.sh `software-properties-common` installation is required to add an apt-key and then later install docker engine. If you want the package removed, you may remove it in build_debian.sh immediately after the usage.

Since the package impact to python applications is unexpected, please also add a big comment block there with your insight here.


---
In reply to: [569194019](https://github.com/Azure/sonic-buildimage/pull/6273#discussion_r569194019) [](ancestors = 569194019,568892398)

Reply:
As suggested, have moved the uninstallation of 'python3-gi' to build_debian.sh.